### PR TITLE
Add h5pickle recipe

### PR DESCRIPTION
**Pull request by Exteris**

### CODE REVIEW

File: recipes/h5pickle/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    - cachetools
+
+test:
+  imports:
+
+about:
+  home: http://github.com/exteris/h5pickle
+  license: MIT
+  license_family: MIT
+  license_file: '{{ environ["RECIPE_DIR"] }}/LICENSE'
```
Comment (jakirkham) [inline recipes/h5pickle/meta.yaml]: This is fine for now. To fix it long term, please see issue ( https://github.com/Exteris/h5pickle/issues/2 ).

File: recipes/h5pickle/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    - setuptools
+  run:
+    - python
+    - h5py
+    - toolz
+    - cachetools
+
+test:
+  imports:
+
```
Comment (jakirkham) [inline recipes/h5pickle/meta.yaml]: We need an import test here. Just `- h5pickle` would suffice.

**Comment by jakirkham** [comment]
Thanks @Exteris. Needs a test, but otherwise LGTM.

File: recipes/h5pickle/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    - toolz
+    - cachetools
+
+test:
+  imports:
+    - h5pickle
+
+about:
+  home: http://github.com/exteris/h5pickle
+  license: MIT
```
Comment (jakirkham) [inline recipes/h5pickle/meta.yaml]: Please add `license_file: LICENSE`.

**Comment by Exteris** [comment]
Thanks for walking me through this @jakirkham

File: recipes/h5pickle/meta.yaml (hunk 3 above)
Comment (jakirkham) [reply recipes/h5pickle/meta.yaml]: note: Done below.

File: recipes/h5pickle/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  license_family: MIT
+  license_file: LICENSE
+  summary: 'Wrapper for h5py with pickle capabilities'
+
+  # The remaining entries in this section are optional, but recommended
```
Comment (jakirkham) [inline recipes/h5pickle/meta.yaml]: nit: We can drop this comment.

**Comment by jakirkham** [comment]
np. Thanks for putting this together @Exteris.

One minor comment above. Otherwise LGTM.

**Comment by jakirkham** [comment]
No need to wait for the other CIs as this is noarch and will only build on CircleCI.

Thanks @Exteris!

**Pull request merged**